src/s3fs.cpp

```diff
@@ -4540,14 +4545,33 @@ static int my_fuse_opt_proc(void* data, const char* arg, int key, struct fuse_ar
       passwd_file = strchr(arg, '=') + sizeof(char);
       return 0;
     }
+    if(0 == strcmp(arg, "ibm_auth")){
+      S3fsCurl::SetIsIBMAuth(true);
+      S3fsCurl::SetIAMCredentialsURL("https://iam.bluemix.net/oidc/token");
+      S3fsCurl::SetIAMTokenField("access_token");
+      S3fsCurl::SetIAMExpiryField("expiration");
+      S3fsCurl::SetIAMFieldCount(2);
+      // IBM auth only supports a-xmz-acl: public-read
+      if(S3fsCurl::SetDefaultAcl("") == "public-read"){
```

Comment:
We need to add `GetDefaultAcl` since this call will set `default_acl` to "" even if it was previously non-empty.  Also I think you want to check this outside of `my_fuse_opt_proc` in case users specify IBM auth before ACL.